content/vgg-classifier-tutorial.md

```diff
@@ -0,0 +1,320 @@
+
+```python
+mymodel.layers[-1] = Dense(101, activation='softmax'))
+```
+and reattached it to the before last layer:
+
+```python
+mymodel.layers[-1].set_previous(pretrained_model.layers[-2])
+```
+Nothing more !
+
+### Lock some layers for training
+Because ImageNet is a giant dataset compare to Caltech, the VGG16 pretrained model already know how to recognize the image details, but the model must be adapted/tuned to this dataset. A simple way to do this is to lock  all the convolution layers during the training, meaning the weights of the convolutional layers are never be update. Only the fully connected layers (``Dense``) are updated. But you can easily make some tests by changing the value of: ``unlock_last``. Increasing ``unlock_last`` means: more layers with by updated, decreasing ``unlock_last`` means: less layers are updated.
+
+In keras, it's easy! You just need to iterate on mymodel.layers to set ``trainable`` attribute of the layer to False or True:
+
+```python
+unlock_last = 5
+for num in range(0,len(mymodel.layers)):
+  layer = mymodel.layers[num]
+  if num < len(mymodel.layers)-unlock_last:
+    layer.trainable = False # Locked for the Firsts
+  else:
+    layer.trainable = True # Unlocked for the Lasts
+```
+### Compile the model
+Now, the model can be compile with the categorical cross-entropy loss function  (typically of classification model).
+Note that the learning rate need to be low, else the system is not able to adapted the model for your dataset.
+If it doesn't work for your own dataset, try to decrease first your the learning rate ``lr`` or next increase it (i.e. 0.05) or change the optimizer by `AdaGrad`, `AdaDelta`...
+```python
+sgd = SGD(lr=0.01, decay=1e-6, momentum=0.9, nesterov=True)
+mymodel.compile(optimizer=sgd, loss='categorical_crossentropy')
+```
+### Train the model
+With keras, one function is enough for doing the training, with couple of parameters:
+* X_data, Y_data are the dataset and the label
+* batch_size=64 is the batch size, if you have a GPU with a small memory size reduce it, with 12Gb you can easily increase to 128
+* validation_split=0.1, is the part of the data used for validate your model, this part is very important, you can only say your model is good enough if the accuracy of the validation is good!
+* show_accuracy=True, Display epoch by epoch information
+* shuffle=True, to shuffle the data at each epoch
+
+And at the end save the new model weights.
+
+```python
+mymodel.fit(X_data, Y_data,
+      batch_size=64,
+      validation_split=0.1,
+      nb_epoch=300,
+      show_accuracy=True,
+      shuffle=True)
+
+mymodel.save_weights('my_model_weights.h5')
+```
+
+If you respect this tutorial, you can quickly achieve an accuracy higher than 85% on caltech101 in 30 epochs, but don't hesitate to play with the different parameters of the model, like the batch_size, learning_rate... to improve the final validation accuracy.
```

Comment:
"If you *follow* this tutorial"

The parameters I would suggest tuning would be the size and number of FC layers after the frozen convolution layers.

I think we need to introduce the possibility of fine-tuning the entire network (after de-freezing the convolution layers) with a very low learning rate, to improve final accuracy. The effect of this is huge.